code/src/esp_32.cpp

```diff
@@ -0,0 +1,29 @@
+#include <esp_32.hpp>
+
+namespace r2d2::communication {
+
+    esp_32_c::esp_32_c(hwlib::spi_bus &spi_connection, hwlib::pin_out &ss)
+        : spi_connection(spi_connection), slave_select(ss) {
+    }
+
+    void esp_32_c::send(const r2d2::frame_external_s &frame) {
+        uint8_t recv[256] = {0};
```

Comment:
In C++, `= { 0 }` on an array will only set the first element of the array to zero. You will want `= {}`.

Reply:
I only  need the fist element of the array to be zero, this is the length of the external frame, if this is zero I can assume I received nothing

Reply:
Could you clarify that in a comment?

Reply:
```c++
struct frame_external_s {
        uint8_t length;
        external_id_s id;
        frame_type type;

        // NOTE: data should come last; ordering is important
        // for this specific struct!
        uint8_t data[256 - sizeof(uint8_t) - sizeof(external_id_s) - sizeof(frame_type)];
};
```
See, the first byte of frame_external_s is its length, an frame without a length contains nothing or did I misunderstand? 
(https://github.com/R2D2-2019/communication/blob/54129b0274da275c574afbf04e155042d91d616a/code/src/esp_32.cpp#L15) 
Here I check whether the first byte received is zero, if zero i do not write the frame into the que

Reply:
I meant in the code, to clarify that you only want the first byte :-P

Reply:
aha